### PR TITLE
Port ting_new_materials to SAL. BBS-177

### PR DESCRIPTION
**Pull request by tripox**

### CODE REVIEW

File: modules/ting_new_materials/ting_new_materials.module [hunk 1]
```diff
@@ -150,25 +152,22 @@ function ting_new_materials_make_limit_query($limit = 0) {
...
   // Search the data well.
   $search_result = array();
   try {
-    // TODO: Consider porting this to the Search Abstraction Layer.
-    $search_result = opensearch_do_search($query, $page + 1, $num_per_page, $options);
+    $sal_query = ting_start_query()
+      ->setFullTextQuery($query)
+      ->setPage($page + 1)
+      ->setCount($num_per_page)
+      ->addSort('date', 'descending')
```
Comment (tripox) [inline modules/ting_new_materials/ting_new_materials.module]: Should this be `->addSort('date_descending', 'descending')`?

File: modules/ting_new_materials/ting_new_materials.module [hunk 2]
```diff
@@ -150,25 +153,22 @@ function ting_new_materials_make_limit_query($limit = 0) {
...
   try {
-    // TODO: Consider porting this to the Search Abstraction Layer.
-    $search_result = opensearch_do_search($query, $page + 1, $num_per_page, $options);
+    $sal_query = ting_start_query()
+      ->setFullTextQuery($query)
+      ->setPage($page + 1)
+      ->setCount($num_per_page)
+      ->addSort('date', TingSearchSort::DIRECTION_DESCENDING)
+      ->addFieldFilter('enrich', TRUE)
+      ->addFieldFilter('collectionType', 'manifestation');
```
Comment (kasperg) [inline modules/ting_new_materials/ting_new_materials.module]: As mentioned in another PR: Are `enrich` and `collectionType` valid field filters?

File: modules/ting_new_materials/ting_new_materials.module [hunk 3]
```diff
@@ -150,25 +153,22 @@ function ting_new_materials_make_limit_query($limit = 0) {
  * @param int $num_per_page
  *   The page number.
  *
- * @return TingClientSearchResult
+ * @return TingSearchResultInterface
```
Comment (kasperg) [inline modules/ting_new_materials/ting_new_materials.module]: This does not seem right. Obviously the return type of this method changes but I would expect that all the callers needs to be updated as well to use the methods provided by the new interface.

**Review by kasperg — CHANGES_REQUESTED** [review]
Reviewed. I have some comments.

**Comment by tripox** [comment]
@kasperg, do you know what 'collectionType' is and why it's set to: 'manifestation'?
Should we support it?

**Comment by kasperg** [comment]
In short `collectionType=manifestation` will make OpenSearch behave as if all collections only contain a single object. In Ding2 this is the same behavior as a search provider which does not support collections would have.

There is more information in [the documentation](https://opensource.dbc.dk/services/open-search-web-service) under the headline Work structure.

**Review by kasperg — APPROVED** [review]

**Pull request merged**